lib/elixir_auth_google.ex

```diff
@@ -8,33 +8,54 @@ defmodule ElixirAuthGoogle do
   @google_token_url "https://oauth2.googleapis.com/token"
   @google_user_profile "https://www.googleapis.com/oauth2/v3/userinfo"
 
+
+  @doc """
+  `get_baseurl_from_conn/1` derives the base URL from the conn struct
+  """
+  @spec get_baseurl_from_conn(Map) :: String.t
+  def get_baseurl_from_conn(conn) do
```

Comment:
I think we could use pattern matching for the parameters and a guard to test the value of the host:
```suggestion
  def get_baseurl_from_conn(%{host: h, port: p}) when host == "localhost" do
    "http://#{h}:#{p}"
  end
  
  def get_baseurl_from_conn(%{host: h}) do
     "https://#{h}"
  end
```

Reply:
That's a much more "elixiry" approach. Thanks! 👍